src/classes/NPSP_Batch.cls

```diff
@@ -0,0 +1,94 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Batch entity class in NPSP.
+*/
+
+public inherited sharing class NPSP_Batch {
+
+    private Id batchId;
+    private List<DataImport__c> dataImports;
+    private List<String> groupIds = new List<String>();
+
+
+    private ElevateCaptureGroupService captureGroupService {
+        get {
+            if (captureGroupService == null) {
+                captureGroupService = new ElevateCaptureGroupService();
+            }
+            return captureGroupService;
+        }
+        set;
+    }
+
+    private void updatePaymentStatusesFor(List<DataImport__c> dataImports,
+            List<ElevateCaptureGroupResponse> captureGroupResponses) {
+        for (DataImport__c dataImport : dataImports) {
+            for (ElevateCaptureGroupResponse response : captureGroupResponses) {
+                if(isValidCaptureResponse(response, dataImport)) {
+                    dataImport.Payment_Status__c = response.status;
+                }
+            }
+        }
+        update dataImports;
+    }
+
+    private Boolean isValidCaptureResponse (ElevateCaptureGroupResponse response, DataImport__c dataImport) {
+        return response.id.equals(dataImport.Payment_Elevate_Batch_ID__c) && response.errors == null;
+    }
+
+    private void setElevateBatchIds() {
```

Comment:
I would suggest:

1. Move the query into a selector
2. Make the selector injectable
3. Establish a "Gifts" domain class
4. Add a filter method to Gifts to return an instance with the Authorized Gifts only
5. Rename the method "captureGroupIds"

![2021-04-30_12-47-02](https://user-images.githubusercontent.com/46458081/116747123-4f060700-a9b2-11eb-9dc4-5ba8f7ac76c5.png)
![2021-04-30_12-47-19](https://user-images.githubusercontent.com/46458081/116747138-5200f780-a9b2-11eb-9ef5-cdc29f22b199.png)
![2021-04-30_12-47-34](https://user-images.githubusercontent.com/46458081/116747148-56c5ab80-a9b2-11eb-9900-87da447e3826.png)